gui.go

```diff
@@ -934,3 +965,10 @@ func (g *Gui) isBlacklisted(k Key) bool {
 	}
 	return false
 }
+
+// busy updates the idle timer with activity
+func (g *Gui) busy() {
```

Comment:
Is this something we can do?:
```go
func (g *Gui) busy() func() {
  g.mu.Lock()
  now := time.Now()
  g.idleTime = time.Now()
  g.idle = false
  g.mu.Unlock()
  return func() {
    g.mu.Lock()
    if g.idleTime == now {
      g.idle = true
    }
    g.mu.Unlock()
  }
}
```

Then we can use busy like so
```go
func a() {
  setIdle := busy()
  defer setIdle()
  doStuff()
}
```

I think this could eliminate the timeout, what do you think?

Reply:
Not sure I follow - each thread doing stuff would assume the gui is idle once it's done? wouldn't that end up having false positives, as one thread is done doing stuff, but another is still working?